app/src/main/java/openfoodfacts/github/scrachx/openfood/fragments/ProductPhotosFragment.java

```diff
@@ -64,72 +55,56 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
             state = ProductFragment.mState;
         }
         product = state.getProduct();
-        mFragment = this;
         // initialize the arraylist
         imageNames = new ArrayList<>();
         imagesRecycler = view.findViewById(R.id.images_recycler);
-        imgMap = new HashMap<>();
 
+        openFoodAPIClient.getImages(product.getCode(), (value, response) -> {
 
-        openFoodAPIClient.getImages(product.getCode(), new OpenFoodAPIClient.OnImagesCallback() {
-            @Override
-            public void onImageResponse(boolean value, String response) {
+            if (value && response != null) {
 
-                if (value && response != null) {
+                // a json object referring to base json object
+                JSONObject jsonObject = Utils.createJsonObject(response);
 
-                    // a json object referring to base json object
-                    JSONObject jsonObject = Utils.createJsonObject(response);
-
-                    // a json object referring to images
-                    JSONObject images = null;
-                    try {
-                        images = jsonObject.getJSONObject("product").getJSONObject("images");
-                    } catch (JSONException e) {
-                        e.printStackTrace();
-                    }
-
-                    if (images != null) {
+                // a json object referring to images
+                JSONObject images = null;
+                try {
+                    images = jsonObject.getJSONObject("product").getJSONObject("images");
+                } catch (JSONException e) {
+                    Log.w(ProductPhotosFragment.class.getSimpleName(), "can get product / images in jon", e);
```

Comment:
`Log.w(ProductPhotosFragment.class.getSimpleName(), "can get product / images in jon", e);`